### ISSUE TITLE
[Accessibility] Dialog has no Accessible Name

### ISSUE THREAD

**Issue by taparikh** [OfficeDev/office-ui-fabric-react#9793]
## Codepen demonstrating the issue
https://codepen.io/Tanvi_p/pen/BgEqda

## Actual behavior:
On running the Accessibility Insights Tool for when a modal popup is launched, the following test fails - 
-Custom widgets: Design Pattern :
 Widget role: dialog
 Design pattern: Dialog (Modal)
 Accessible name: (no value)
Also a red line appears on the screen when visual helper for this component is activated. 

## Expected behavior:
Dialog should have an accessible name

## Documentation describing expected behavior
We have included all the aria props as provided by the modal fabric component. Please see screenshots attached for more clarity.

![dialog_error](https://user-images.githubusercontent.com/51679982/61156650-c0f12180-a4a8-11e9-92d9-2717183d4054.JPG)

**Comment by Vitalius1** [comment]
@taparikh , the codepen you provided does not work for me and if honestly has way too much stuff in it to allow to isolate any possible issue with Dialog accessibility. I verified the Dialogs on our [demo website](https://developer.microsoft.com/en-us/fabric#/controls/web/dialog) with Narrator on and it seems that Narrator doesn’t have any issues announcing that a Dialog was open and it reads out the Heading and the content inside of it. It might be that Accessibility Insights Tool checks a specific html element and flags it, but I don’t see any issues with screen reader’s readout. Can you please verify an actual accessibility pass with an AT (like Narrator) and report back the results. I feel like a static analysis by your tool might flag some false positives. Thanks!

**Comment by Vitalius1** [comment]
Also, we have an a11y pass report tracked in #8061, but I think a few of the points in the description are not issues anymore.

**Comment by taparikh** [comment]
@Vitalius1  Thanks for your note - We will test it using a narrator & will get back to you if any questions

**Comment by taparikh** [comment]
@Vitalius1 Hi I installed the JAWS screen reader and tested the popup modal in our environment.  It did announce that the dialog was open but it also went head and read the menu options on the table header. I am not sure if this is expected behavior. Id be happy to hop on a quick teams call to discuss this with you

**Comment by Vitalius1** [comment]
If you go on our demo website on Dialog page to test it you will notice that it reads out the heading and then all the content inside of that Dialog, so I would expect the fact that in your environment reading the table header is expected. I would also be very thankful if you could provide a repro with an isolated Dialog or Modal component so I can understand what is the issue.

**Comment by taparikh** [comment]
@Vitalius1 As much as id like to share an isolated version, the dialog is closely tied to the data layer & its existence is conditional. I think at the point it would be helpful for us to hop on a teams call & debug this issue (and possibly others too) in our local environment. I'd happy to share dev URL as well with a couple days access so you can check the bugs.

**Comment by Vitalius1** [comment]
@taparikh When going to our [demo page for `Dialog`](https://developer.microsoft.com/en-us/fabric#/controls/web/dialog) and run the Accessibility Insights on it you can see that the issue you are reporting is no longer there. When looked at the example code I can see that the fix consists in adding hidden from view labels and assigning to the a unique ID that is later passed into `modalProps` prop. See code here:
https://github.com/OfficeDev/office-ui-fabric-react/blob/35e8857c18f8eac3cc3b75cd0353b9bfec3128b3/packages/office-ui-fabric-react/src/components/Dialog/examples/Dialog.Basic.Example.tsx#L37-L42

https://github.com/OfficeDev/office-ui-fabric-react/blob/35e8857c18f8eac3cc3b75cd0353b9bfec3128b3/packages/office-ui-fabric-react/src/components/Dialog/examples/Dialog.Basic.Example.tsx#L53-L54

**Comment by taparikh** [comment]
@Vitalius1 We are still seeing an empty accessible name for the dialog as shown in the screenshot below

![image](https://user-images.githubusercontent.com/51679982/61670695-70907580-ac99-11e9-90d8-c1809ebfb4a0.png)

Below you can see the HTML elements with the modal highlighted & all the accessibility properties being empty
![image](https://user-images.githubusercontent.com/51679982/61670755-b5b4a780-ac99-11e9-8660-bc396ebf6d5f.png)

We would like to help you close on this as soon as possible. Can we hop on a call to demo & discuss this in detail ?

**Comment by taparikh** [comment]
Discussed in detail with @Vitalius1 over a call & issue has been resolved.

**Comment by taparikh** [comment]
@Vitalius1 - tested out the snippet on another modal today & it does not work. Both the accessibility insights tool & narrator (using buddy as well) gives an empty document name for accessible name.

**Comment by taparikh** [comment]
Retried without the typo error & works as expected. Closing this issue!
